File: backend-ruby/app/models/ky_thi.rb
class KyThi < ApplicationRecord
  self.table_name = 'KyThi'
  self.primary_key = 'KyThiID'
  belongs_to :mon_thi, class_name: "MonThi", foreign_key: "MonThiID"
  has_many :dang_ky_thi_records, class_name: "DangKyThi", foreign_key: "KyThiID"
  has_many :xep_cho_records, class_name: "XepCho", foreign_key: "KyThiID"
  has_many :phan_phong_records, class_name: "PhanPhong", foreign_key: "KyThiID"
  has_many :diem_danh_records, class_name: "DiemDanh", foreign_key: "KyThiID"
end

File: backend-csharp/Models/KyThi.cs
using System.Text.Json.Serialization;

namespace backend_csharp.Models;

public class KyThi
{
    public int KyThiID { get; set; }
    public string MaKyThi { get; set; } = string.Empty;
    public string TenKyThi { get; set; } = string.Empty;
    public int MonThiID { get; set; }
    public DateOnly NgayThi { get; set; }
    public TimeOnly GioBatDau { get; set; }
    public TimeOnly GioKetThuc { get; set; }
    public DateTime? ThoiHanDangKyDen { get; set; }
    public string TrangThai { get; set; } = "draft";
    public string? MoTa { get; set; }
    public DateTime TaoLuc { get; set; }
    public DateTime? CapNhatLuc { get; set; }
    public MonThi? MonThi { get; set; }
    [JsonIgnore] public ICollection<DangKyThi> DangKyThiRecords { get; set; } = [];
    [JsonIgnore] public ICollection<XepCho> XepChoRecords { get; set; } = [];
    [JsonIgnore] public ICollection<PhanPhong> PhanPhongRecords { get; set; } = [];
    [JsonIgnore] public ICollection<DiemDanh> DiemDanhRecords { get; set; } = [];
}
